MerlinAutoUpdate.sh

```diff
@@ -128,26 +134,40 @@ Update_Custom_Settings() {
     esac
 }
 
+##----------------------------------------##
+## Modified by Martinski W. [2023-Oct-05] ##
+##----------------------------------------##
 get_current_firmware() {
-    local current_version="$(nvram get buildno)_$(nvram get extendno)"
+    local current_version="$(nvram get buildno).$(nvram get extendno)"
     echo "$current_version"
 }
 
+##----------------------------------------##
+## Modified by Martinski W. [2023-Oct-05] ##
+##----------------------------------------##
 get_latest_firmware() {
     local url="$1"
-    local page=$(curl -s "$url")
 
-    local links_and_versions=$(echo "$page" | grep -o 'href="[^"]*'"$MODEL"'[^"]*\.zip' | sed 's/amp;//g; s/href="//' | 
-        awk -F'[_\.]' '{print $3"."$4"."$5" "$0}' | sort -t. -k1,1n -k2,2n -k3,3n)
+    local links_and_versions="$(curl -s "$url" | grep -o 'href="[^"]*'"$MODEL"'[^"]*\.zip' | sed 's/amp;//g; s/href="//' | 
+        awk -F'[_\.]' '{print $3"."$4"."$5" "$0}' | sort -t. -k1,1n -k2,2n -k3,3n)"
+
+    if [ -z "$links_and_versions" ]
+    then echo "**ERROR** **NO_URL**" ; return 1 ; fi
 
-    local latest=$(echo "$links_and_versions" | tail -n 1)
-    local version=$(echo "$latest" | cut -d' ' -f1 | awk -F. '{print $(NF-1)"."$NF}')
-    local link=$(echo "$latest" | cut -d' ' -f2-)
+    local latest="$(echo "$links_and_versions" | tail -n 1)"
+    local linkStr="$(echo "$latest" | cut -d' ' -f2-)"
+    local fileStr="$(echo "$linkStr" | grep -oE "/${MODEL}_[0-9]+.*.zip$")"
+    local versionStr
+
+    if [ -z "$fileStr" ]
```

Comment:
Will quickly test these changes, but i like how it seems to The changes seem to add flexibility to how versions can be derived. This may be necessary if the format or naming convention of the firmware files changes. However, it also introduces more complexity. The decision to modify this part of the script could be based on observed changes in firmware naming conventions or a need to support more varied names.